core/matchers/src/main/kotlin/au/com/dius/pact/core/matchers/Matchers.kt

```diff
@@ -74,6 +78,34 @@ object Matchers : KLogging() {
       resolveMatchers(matchers, category, path, pathComparator).isNotEmpty()
     else false
 
+  /**
+   * Uses concept from "Matchers.INSTANCE.matchesPath('$[*]', ['$', 'str']) == 0"
+   * to determine if an array star matcher is defined on the ancestor
+   */
+  @JvmStatic
+  fun isArrayStarMatcherDefined(path: List<String>, category: String, matchers: MatchingRules?) =
+    if (matchers != null) {
+      val resolvedMatchers = matchers.rulesForCategory(category).filter(Predicate {
+        matchesPath(it, path) == 0
```

Comment:
`matchesPath` is a little confusing, it could use a KDoc.

It returns what index (starting from 1) that a matching expression matches against a path array.  Here are some examples I ran to clarify for myself:
```groovy
    Matchers.INSTANCE.matchesPath('$.a', ['$', 'a']) == 2
    Matchers.INSTANCE.matchesPath('$.a', ['$', 'a', 'b']) == 2
    Matchers.INSTANCE.matchesPath('$.a.b', ['$', 'a', 'b']) == 3
    Matchers.INSTANCE.matchesPath('$.a.c', ['$', 'a', 'b']) == 0 // no match
    Matchers.INSTANCE.matchesPath('$.a.0', ['$', 'a', '0']) == 3
    Matchers.INSTANCE.matchesPath('$.a.0', ['$', 'a', '1']) == 0 // no match
    Matchers.INSTANCE.matchesPath('$.a[*]', ['$', 'a', '0']) == 3
    Matchers.INSTANCE.matchesPath('$.*', ['$', 'a', '0']) == 2
    Matchers.INSTANCE.matchesPath('$[*]', ['$', 'a', '0']) == 0 // because a isn't array index
    Matchers.INSTANCE.matchesPath('$[*]', ['$', '0', '0']) == 2
```

Given this, I don't think using matchesPath would work.  I'm still not sure about a way that would work, other than trying to get matchers for an out-of-bound index.